test/src/main/java/org/apache/accumulo/test/compaction/ExternalDoNothingCompactor.java

```diff
@@ -53,6 +53,9 @@ protected Runnable createCompactionJob(TExternalCompactionJob job, LongAdder tot
       LongAdder totalInputBytes, CountDownLatch started, CountDownLatch stopped,
       AtomicReference<Throwable> err) {
 
+    // Set this to true so that only 1 external compaction is run
+    this.shutdown = true;
```

Comment:
Is this the best place to set shutdown to true? Would it be better to set this right before `throw new CompactionCanceledException();`?

Reply:
This method is called [here](https://github.com/apache/accumulo/blob/main/server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java#L670) to create a Thread, then the Thread is started a few lines later. The shutdown variable is checked in the enclosing while loop, so I don't think it matters in this case.